src/components/MyAccount/NewSettings/SettingsSelectForm.tsx

```diff
@@ -0,0 +1,201 @@
+import { useContext, useState } from "react"
+import { PatronDataContext } from "../../../context/PatronDataContext"
+import {
+  Flex,
+  Select,
+  SkeletonLoader,
+  Text,
+} from "@nypl/design-system-react-components"
+import SettingsLabel from "./SettingsLabel"
+import SaveCancelButtons from "./SaveCancelButtons"
+import type { Patron, SierraCodeName } from "../../../types/myAccountTypes"
+import EditButton from "./EditButton"
+
+interface HomeLibraryNotificationFormProps {
+  type: "library" | "notification"
+  patronData: Patron
+  settingsState
+  pickupLocations: SierraCodeName[]
+}
+
+const HomeLibraryNotificationForm = ({
+  type,
+  patronData,
+  settingsState,
+  pickupLocations,
+}: HomeLibraryNotificationFormProps) => {
+  const { getMostUpdatedSierraAccountData } = useContext(PatronDataContext)
+  const [isLoading, setIsLoading] = useState(false)
+  const [isEditing, setIsEditing] = useState(false)
+  const [error, setError] = useState(false)
+
+  const { setStatus, editingField, setEditingField } = settingsState
+
+  const notificationPreferenceMap =
+    patronData.notificationPreference === "-"
+      ? [
+          { code: "z", name: "Email" },
+          { code: "p", name: "Phone" },
+          { code: "-", name: "None" },
+        ]
+      : [
+          { code: "z", name: "Email" },
+          { code: "p", name: "Phone" },
+        ]
+
+  const validateInput = (input) => {
+    if (input == "Phone" && patronData.phones.length === 0) {
+      setError(true)
+    }
+  }
+
+  const sortedPickupLocations = [
+    patronData.homeLibrary,
+    ...pickupLocations.filter(
+      (loc) => loc.code.trim() !== patronData.homeLibrary.code.trim()
+    ),
+  ]
+
+  const formUtils = {
+    initialState:
+      type === "notification"
+        ? notificationPreferenceMap.find(
+            (pref) => pref.code === patronData.notificationPreference
+          )?.name
+        : patronData.homeLibrary.name,
+    options:
+      type === "notification"
+        ? notificationPreferenceMap
+        : sortedPickupLocations,
```

Comment:
The `sortedPickupLocations` array is not expensive to create, but one concern is having each separate "library" or "notification" select form know about (by creating) both type of data. You can probably pass the data you need from the parent instead of doing both operations here.

It's not a big concern because they're not expensive but something to keep in mind.